### PR TITLE
[WFLY-17510] Convert MicroProfile OpenTracing subsystem to Admin-Only Mode

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-17510

Analysis doc for MP OpenTracing Removal

### CODE REVIEW

File: microprofile/WFLY-17510-remove-microprofile-opentracing.adoc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+* https://issues.redhat.com/browse/WFLY-17156[WFLY-17156]
+* https://issues.redhat.com/browse/EAP7-2002[EAP7-2002]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
```
Comment (fabiobrz) [inline microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:29]: That's me, again :slightly_smiling_face:

File: microprofile/WFLY-17510-remove-microprofile-opentracing.adoc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+=== Testing By
+* [x] Engineering
+* [ ] QE
+
+=== Affected Projects or Components
+
+All the code changes will be confined to the MicroProfile OpenTracing extension and subsystem, which includes the Java code, Galleon modules/layers, etc.
+
```
Comment (fabiobrz) [inline microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:37]: As projects I'd say Wildfly

File: microprofile/WFLY-17510-remove-microprofile-opentracing.adoc [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [x] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+MicroProfile OpenTracing will be migrated to an admin-only module
```
Comment (fabiobrz) [inline microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:52]: " and removed from standard configs" maybe?

File: microprofile/WFLY-17510-remove-microprofile-opentracing.adoc [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+== Backwards Compatibility
+
+This represents the removal of a subsystem, so any application the depends on these APIs will fail to deploy. Likewise, any installation that is expecting (Jaeger) tracing data to be exported will see a behavior change that may or may not impact any external monitoring solutions.
+
+All other subsystems should be unaffected by this removal.
+
+=== Default Configuration
+
+The default configuration for MicroProfile configurations will be altered to remove the MicroProfile OpenTracing subsystem. Likewise, the relevant Galleon layers will also have the extension removed from their definitions.
```
Comment (fabiobrz) [inline microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:70]: Is it Galleon layers or feature packs? I am not sure though, just asking.

File: microprofile/WFLY-17510-remove-microprofile-opentracing.adoc [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+=== Interoperability
+
+See the section of above for a discussion of trace data formats and exports.
+
+== Test Plan
+
+Tests will be added to verify that MicroProfile OpenTracing has been removed, and parsing tests will be modified as needed to verify that the system can continue to read/parse the subsystem config during migration.
+
+Tests for MP OpenTracing under `testsuite/integration/microprofile` and `testsuite/integration/microprofile-tck` will be removed.
```
Comment (fabiobrz) [inline microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:92]: And similarly for the EAP MicroProfile TS, see https://github.com/jboss-eap-qe/eap-microprofile-test-suite

File: microprofile/WFLY-17510-remove-microprofile-opentracing.adoc [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+== Community Documentation
+
+Any documentation related to MicroProfile OpenTracing will be removed.
```
Comment (fabiobrz) [inline microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:96]: s/removed./removed as part of the feature implementation PR./

**Review by fabiobrz — CHANGES_REQUESTED** [review]
Thanks for this proposal @jasondlee!
Just a few minor comments, let me know what you think.

File: microprofile/WFLY-17510-remove-microprofile-opentracing.adoc (hunk 1 above)
Comment (jasondlee) [reply microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:29]: Right! Fixed. :)

File: microprofile/WFLY-17510-remove-microprofile-opentracing.adoc (hunk 3 above)
Comment (jasondlee) [reply microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:52]: Yes. Good catch.

File: microprofile/WFLY-17510-remove-microprofile-opentracing.adoc (hunk 4 above)
Comment (jasondlee) [reply microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:70]: I think, both actually. There are `layer-spec.xml` files, as well as feature pack definitions. @bstansberry can chime in if I'm off base here. :)

File: microprofile/WFLY-17510-remove-microprofile-opentracing.adoc (hunk 5 above)
Comment (jasondlee) [reply microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:92]: Added

File: microprofile/WFLY-17510-remove-microprofile-opentracing.adoc (hunk 4 above)
Comment (bstansberry) [reply microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:75]: It's layers.

The feature pack still incorporates MP OT as a FP incorporates everything that can get provisioned, and MP OT can get provisioned. The galleon content used for the default configs and the layers are what drive whether MP OT shows up in the configuration files. That's what we want to change.

That said, the words "the relevant Galleon layers" should be replaced with an explicit list of the layers that are going to change.

Comment (jasondlee) [reply microprofile/WFLY-17510-remove-microprofile-opentracing.adoc:75]: I'll update that text to include those layers.

Comment (jasondlee) [reply microprofile/WFLY-17510-remove-microprofile-opentracing.adoc]: @bstansberry @fabiobrz Edited the text. Is that better?

Comment (fabiobrz) [reply microprofile/WFLY-17510-remove-microprofile-opentracing.adoc]: Yep, I think it solves Brian's concerns.

**Review by fabiobrz — APPROVED** [review]
Thanks for addressing my comments @jasondlee - the proposal looks good from the QE perspective, approving.

**Comment by jasondlee** [comment]
Great. Thanks for all the help @fabiobrz!

**Pull request merged**